src/object/interrupt.c

```diff
@@ -145,12 +145,20 @@ void invokeIRQHandler_AckIRQ(irq_t irq)
 
 #if defined ENABLE_SMP_SUPPORT && defined CONFIG_ARCH_ARM
     if (IRQ_IS_PPI(irq) && IRQT_TO_CORE(irq) != getCurrentCPUIndex()) {
+#ifdef CONFIG_ARM_GIC_V3_SUPPORT
+        doRemoteDeactivatePrivateInterrupt(IRQT_TO_CORE(irq), IRQT_TO_IDX(irq));
+#else /* CONFIG_ARM_GIC_V3_SUPPORT */
         doRemoteMaskPrivateInterrupt(IRQT_TO_CORE(irq), false, IRQT_TO_IDX(irq));
+#endif /* CONFIG_ARM_GIC_V3_SUPPORT */
         return;
     }
 #endif
+#ifdef CONFIG_ARM_GIC_V3_SUPPORT
+    deactivateInterrupt(irq);
+#else /* CONFIG_ARM_GIC_V3_SUPPORT */
```

Comment:
Same here, having ARM config dependent ifdefs in generic code will make generic proofs impossible -- it's fine for those that we're not planning to ever verify, but GICv3 does sound like something we'd want pretty soon.

Reply:
I think that would help, yes